File: sem6/se/Cocomo.java
import java.util.Scanner;
public class Cocomo {
    public static double effort(double kloc, double[] constants) {
      return constants[0] * (Math.pow(kloc, constants[1]));
  }
  public static double timefordev(double effort, double[] constants) {
      return 2.5 * (Math.pow(effort, constants[2]));
  }
  public static double[] organic(double kloc) {
      double effort = effort(kloc, Constants.organic());
      double time = timefordev(effort, Constants.organic());
      return new double[] { effort, time };
  }
  public static double[] semidetached(double kloc) {
      double effort = effort(kloc, Constants.semidetached());
      double time = timefordev(effort, Constants.semidetached());
      return new double[] { effort, time };
  }
  public static double[] embedded(double kloc) {
      double effort = effort(kloc, Constants.embedded());
      double time = timefordev(effort, Constants.embedded());
      return new double[] { effort, time };
  }
  public static void main(String[] args) {
      Scanner sc = new Scanner(System.in);
      System.out.print("Enter number of KLOC: ");
    int kloc = sc.nextInt();
    if(kloc >=2 && kloc < 50) {
        toString("Organic", organic(kloc));
    } else if (kloc >= 50 && kloc < 300) {
        toString("Semi-detached", semidetached(kloc));
    } else if(kloc >= 300) {
        toString("embedded", embedded(kloc));
    }
    sc.close();
  }
  public static void toString(String mode, double[] results) {
      System.out.println("\nMode: " + mode);
    System.out.println("Effort: " + results[0] + "\nTime: " + results[1]);
  }
}
class Constants {
    public static double[] organic() {
      return new double[] { 2.4, 1.05, 0.38 };
  }
  public static double[] semidetached() {
      return new double[] { 3, 1.12, 0.35 };
  }
  public static double[] embedded() {
      return new double[] { 3.6, 1.2, 0.32 };
  }
}

File: sem6/se/cocomo.c
#include <stdio.h>
#include <math.h>

double effort(double kloc, double constants[]) {
    return constants[0] * pow(kloc, constants[1]);
}

double timefordev(double effort, double constants[]) {
    return 2.5 * pow(effort, constants[2]);
}

void organic(double kloc, double results[]) {
    double constants[] = {2.4, 1.05, 0.38};
    results[0] = effort(kloc, constants);
    results[1] = timefordev(results[0], constants);
}

void semidetached(double kloc, double results[]) {
    double constants[] = {3.0, 1.12, 0.35};
    results[0] = effort(kloc, constants);
    results[1] = timefordev(results[0], constants);
}

void embedded(double kloc, double results[]) {
    double constants[] = {3.6, 1.2, 0.32};
    results[0] = effort(kloc, constants);
    results[1] = timefordev(results[0], constants);
}

void print_results(const char* mode, double results[]) {
    printf("\nMode: %s\n", mode);
    printf("Effort: %f\n", results[0]);
    printf("Time: %f\n", results[1]);
}

int main() {
    double kloc;
    printf("Enter number of KLOC: ");
    scanf("%lf", &kloc);

    double results[2];

    if (kloc >= 2 && kloc < 50) {
        organic(kloc, results);
        print_results("Organic", results);
    } else if (kloc >= 50 && kloc < 300) {
        semidetached(kloc, results);
        print_results("Semi-detached", results);
    } else if (kloc >= 300) {
        embedded(kloc, results);
        print_results("Embedded", results);
    }

    return 0;
}
